include/picongpu/fields/incidentField/profiles/ExpRampWithPrepulse.hpp

```diff
@@ -30,276 +30,290 @@
-                            float_X const t2,
-                            float_X const a2,
-                            float_X const t) const
-                        {
-                            auto const log1 = (t2 - t) * math::log(a1);
-                            auto const log2 = (t - t1) * math::log(a2);
-                            return math::exp((log1 + log2) / (t2 - t1));
-                        }
-                    };
-                } // namespace detail
-            } // namespace profiles
-
-            namespace detail
+                return "ExpRampWithPrepulse";
+            }
+        };
+
+
+        namespace detail
+        {
+            /** Unitless exponential ramp with prepulse parameters (envelope part)
+             *
+             * @tparam T_Params user (SI) parameters
+             */
+            template<typename T_Params>
+            struct ExpRampWithPrepulsesLongitudinalUnitless : public BaseParamUnitless<T_Params>
+            {
+                //! User SI parameters
+                using Params = T_Params;
+
+                using Base = BaseParamUnitless<T_Params>;
+
+                // unit: UNIT_TIME
+                static constexpr float_X LASER_NOFOCUS_CONSTANT
+                    = static_cast<float_X>(Params::LASER_NOFOCUS_CONSTANT_SI / UNIT_TIME);
+
+                static constexpr float_X TIME_PREPULSE = static_cast<float_X>(Params::TIME_PREPULSE_SI / UNIT_TIME);
+                static constexpr float_X TIME_PEAKPULSE = static_cast<float_X>(Params::TIME_PEAKPULSE_SI / UNIT_TIME);
+                static constexpr float_X TIME_1 = static_cast<float_X>(Params::TIME_POINT_1_SI / UNIT_TIME);
+                static constexpr float_X TIME_2 = static_cast<float_X>(Params::TIME_POINT_2_SI / UNIT_TIME);
+                static constexpr float_X TIME_3 = static_cast<float_X>(Params::TIME_POINT_3_SI / UNIT_TIME);
+                static constexpr float_X endUpramp = TIME_PEAKPULSE - 0.5_X * LASER_NOFOCUS_CONSTANT;
+                static constexpr float_X startDownramp = TIME_PEAKPULSE + 0.5_X * LASER_NOFOCUS_CONSTANT;
+
+
+                // compile-time checks for physical sanity:
+                static_assert(
+                    (TIME_1 < TIME_2) && (TIME_2 < TIME_3) && (TIME_3 < endUpramp),
+                    "The times in the parameters TIME_POINT_1/2/3 and the beginning of the plateau (which is "
+                    "at "
+                    "TIME_PEAKPULSE - 0.5*RAMP_INIT*PULSE_DURATION) should be in ascending order");
+
+                // some prerequisites for check of intensities (approximate check, because I can't use exp and
+                // log)
+                static constexpr float_X ratio_dt
+                    = (endUpramp - TIME_3) / (TIME_3 - TIME_2); // ratio of time intervals
+                static constexpr float_X ri1
+                    = Params::INT_RATIO_POINT_3 / Params::INT_RATIO_POINT_2; // first intensity ratio

```

Comment:
Could you please add a compile time assert to check if `INT_RATIO_POINT_2` ≠ 0 (or even >0)?

Reply:
Code that existed before